servicetalk-http-api/src/main/java/io/servicetalk/http/api/StreamingHttpPayloadHolder.java

```diff
@@ -222,6 +223,9 @@ private void updatePayloadBody(Publisher<Buffer> newPayload, boolean isTransform
                 // payloadBody() will split trailers if not yet split
                 newPayload.liftSync(new BridgeFlowControlAndDiscardOperator(payloadBody()));
         payloadInfo.setOnlyEmitsBuffer(true);
+        if (isTransform) {
+            payloadInfo.setSafeToAggregate(false);
```

Comment:
When a request/response is transformed then we have no info of the new payload, thus we revert safe-to-aggregate back to false.

Reply:
Discussed offline: users of aggregated API usually assume that the payload body always fits in memory. Transform functions are usually used for logging/metrics/decoration. If they have a use-case when transformation changes the size significantly, they have to handle it using streaming API.